Starting point for CONTRIBUTORS.md? (#9)
LittleBigRefresh/Refresh: CONTRIBUTING.md

# New Contributor's Survival Guide

> This document details the steps and general tips on how to contribute to the Refresh private server software! Feel free to skip sections if you believe you're already set.

Reviewer: Looks a little weird stylized that way

```suggestion
This document details the steps and other important information on how to contribute to Refresh. Feel free to skip sections if you believe you're already set.
```

> ## Requirements:
> - Knowledge of the [.NET 7 SDK](https://learn.microsoft.com/dotnet/)
> - Ability to learn [Realm](https://realm.io), an easy to pick up database software
> - Basic knowledge of "reverse engineering," specifically reading packet captures and (rarely) reading the game executable for the LBP series.

Reviewer: ```suggestion
## Requirements:
- Basic knowledge of [.NET](https://learn.microsoft.com/dotnet/)
- Ability to work with [Realm](https://realm.io) if necessary
- Basic knowledge of "reverse engineering," specifically reading packet captures and (rarely) reading the game executable for the LBP series.
```

## Software prerequisites
- [.NET 7 SDK](https://dotnet.microsoft.com)
- [Git](https://git-scm.com)

> - An IDE that supports .NET; Visual Studio, Visual Studio Code, JetBrains Rider, Eclipse, it's up to you!

Reviewer: I'm sorry but I wouldn't do that to my worst enemy
```suggestion
- An IDE that supports .NET; Visual Studio, Visual Studio Code, JetBrains Rider, it's up to you!
```

- Any legally acquired PS3 or PSV LBP game, excluding Karting.
- Access to [HEN](https://www.psx-place.com/threads/tutorial-ps3hen-the-great-ps3-hen-all-in-one-guide.24369/)/CFW (PS3), [Henkaku/Enso](https://vita.hacks.guide) (PSV), and/or [RPCS3](https://rpcs3.net) (PS3 on PC).

# Preparing your system
In order to use Git you must set up your configuration, this should ideally match your GitHub account's name and email. This will modify the **global** config, which will allow you to contribute to multiple projects with the same name and email with ease.

`$ git config --global user.name Your Name`

> `$ git config --global user.email you@example.com`

Reviewer: I think it would be better to forward newbies towards GitHub Desktop. I've not used it myself but I've had good experiences from telling people to use it.

Author: I've changed the wording on that section to include both CLI and GitHub Desktop. LMK your thoughts.

Reviewer: Looks great!

# Preparing your new development environment
It's almost time to clone Refresh! Create a fork by pressing the "Fork" button at the base of this repository.

> Afterwards, Open a terminal to your working directory and clone the new fork using Git.

Reviewer: ```suggestion
Afterwards, open a terminal to your working directory and clone the new fork using Git.

This is usually done with the following command:
` $ git@github.com:<YOUR_USERNAME>/Refresh.git`
```

> Restore the dependencies needed for development after changing directory to the new fork.
> 
> `$ dotnet restore`

Reviewer: This will be done automatically by IDEs and the dotnet utility later on. No need to have people worrying about this.
```suggestion
```

## Software prerequisites
- [.NET 7 SDK](https://dotnet.microsoft.com)
- [Git](https://git-scm.com)
- An IDE that supports .NET; Visual Studio, Visual Studio Code, JetBrains Rider, Eclipse, it's up to you!
- Any legally acquired PS3 or PSV LBP game, excluding Karting.
- Access to [HEN](https://www.psx-place.com/threads/tutorial-ps3hen-the-great-ps3-hen-all-in-one-guide.24369/)/CFW (PS3), [Henkaku/Enso](https://vita.hacks.guide) (PSV), and/or [RPCS3](https://rpcs3.net) (PS3 on PC).

# Preparing your system
In order to use Git you must set up your configuration, this should ideally match your GitHub account's name and email. This will modify the **global** config, which will allow you to contribute to multiple projects with the same name and email with ease.

`$ git config --global user.name Your Name`

`$ git config --global user.email you@example.com`

# Preparing your new development environment
It's almost time to clone Refresh! Create a fork by pressing the "Fork" button at the base of this repository.

Afterwards, Open a terminal to your working directory and clone the repository using Git.

> `$ git clone https://github.com/LittleBigRefresh/Refresh`

Reviewer 2: Shouldnt the user be pointed to clone their fork and not the main repo? or are there some commands here missing to set the remote URLs?

Reviewer 2: In an ideal development environment, you have 2 remotes, one for your fork (generally called `origin`) and another remote called upstream (that would be `https://github.com/LittleBigRefresh/Refresh` in this case)

Author: > In an ideal development environment, you have 2 remotes, one for your fork (generally called `origin`) and another remote called `upstream` (that would be `https://github.com/LittleBigRefresh/Refresh` in this case)

Cloning the fork should theoretically do this for you

Reviewer 2: > > In an ideal development environment, you have 2 remotes, one for your fork (generally called `origin`) and another remote called `upstream` (that would be `https://github.com/LittleBigRefresh/Refresh` in this case)
> 
> Cloning the fork should theoretically do this for you

It does not do that, you have to add the `upstream` remote manually

Author: That is correct, sorry about that
